### PR TITLE
expression: return NULL when error is not nil

### PR DESCRIPTION
**Pull request by zz-jason**

### CODE REVIEW

**Comment by zz-jason** [comment]
@XuHuaiyu PTAL

File: expression/builtin_encryption.go [hunk 1]
```diff
@@ -228,7 +228,7 @@ func (b *builtinPasswordSig) evalString(row []types.Datum) (d string, isNull boo
 	sc := b.ctx.GetSessionVars().StmtCtx
 	pass, isNull, err := b.args[0].EvalString(row, sc)
 	if isNull || err != nil {
-		return "", false, errors.Trace(err)
+		return "", err != nil, errors.Trace(err)
```
Comment (coocood) [inline expression/builtin_encryption.go]: Should be `return "", true, errors.Trace(err)`?

Comment (zz-jason) [reply expression/builtin_encryption.go]: no, `password(NULL)` should return an empty string, not `NULL`

**Comment by coocood** [comment]
LGTM

**Pull request merged**